modules/opensearch/src/OpenSearchTingObject.php

```diff
@@ -31,26 +32,73 @@ class OpenSearchTingObject implements TingObjectInterface {
   protected $openSearchObject;
 
   /**
-   * @var string Ding specific ID for the object. Usually the entity id.
+   * @var string Id of the owner of the object, eg. the Agency.
    */
-  protected $dingId;
+  protected $ownerId;
 
   /**
-   * @var string Id of the owner of the object, eg. the Agency.
+   * @var TingRelation[] list of materials related to this material.
    */
-  protected $ownerId;
+  protected $relations;
+
+  /**
+   * OpenSearchObject constructor.
+   *
+   * @param TingClientObject $open_search_object
+   *   The Open Search result this object wraps.
+   */
+  public function __construct($open_search_object) {
+    $this->openSearchObject = $open_search_object;
+  }
 
   /**
    * {@inheritdoc}
    */
   public function getId() {
-    return $this->dingId;
+    return $this->openSearchObject->id;
   }
 
-  public function getRelations(){
-    return [];
+  /**
+   * {@inheritdoc}
+   */
+  public function getRelations() {
+    // If relations are not set; do another request to get relations.
+    if (!isset($this->openSearchObject->relationsData)) {
+      $ting_client_object = ting_get_object($this->ding_entity_id, FALSE, TRUE);
+      if (isset($ting_client_object->relationsData)) {
+        $this->openSearchObject->relationsData = $ting_client_object->relationsData;
+      }
+    }
+
+    $this->openSearchObject->relations = array();
+    $relation_objects = array();
+    if (isset($this->openSearchObject->relationsData)) {
+      $entity_ids = array();
+      foreach ($this->openSearchObject->relationsData as $record) {
+        if (isset($record->relationUri) && isset($record->relationObject)) {
+          $entity_ids[] = $record->relationUri;
+        }
+      }
+      if ($entity_ids) {
```

Comment:
Perhaps `!empty($entity_ids)` here to reduce implicit type juggling.

Reply:
Will refactor ...